packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -263,7 +263,7 @@ class AndroidDevice extends Device {
     if (route != null)
       cmd.addAll(<String>['--es', 'route', route]);
     if (options.debuggingEnabled) {
-      if (options.checked)
```

Comment:
We still need to respect the `--checked` / `--no-checked` flag for a little bit longer, until the various build modes are further plumbed through the system (context: https://github.com/flutter/flutter/issues/3850).